LocationSync/LocationSync.cpp

```diff
@@ -164,5 +184,127 @@ namespace Plugin {
         }
     }
 
+        void LocationSync::onNetControlTimer()
+        {
+            static uint8_t remainingAttempts = _retries;
+            bool networkReachable = getConnectivity();
+            remainingAttempts--;
+            TRACE(Trace::Information, (_T("Network is %s"), networkReachable ? "REACHABLE" : "UNREACHABLE"));
+            if (networkReachable || remainingAttempts <= 0)
+            {
+                _netControlTimer.stop();
+                TRACE(Trace::Information, (_T("Network reachability monitoring stopped.")));
+                TRACE(Trace::Information, (_T("Proceeding with LocationService init.")));
+                _sink.Initialize(_source, _interval, _retries);
+            } else {
+                TRACE(Trace::Information, (_T("Doing one more reachability check in %d sec, remaining attempts: %d"), _interval, remainingAttempts));
+            }
+        }
+        bool LocationSync::getConnectivity()
+        {
+            JsonObject joGetParams;
+            JsonObject joGetResult;
+            std::string callsign = "org.rdk.Network.1";
+            std::string token;
+            /* check if plugin active */
+            if (false == Utils::isPluginActivated("org.rdk.Network")) {
+                TRACE(Trace::Information, ("Network plugin is not activated \n"));
+                return false;
+            }
+
+            Utils::SecurityToken::getSecurityToken(token);
+            string query = "token=" + token;
+            Core::SystemInfo::SetEnvironment(_T("THUNDER_ACCESS"), _T(SERVER_DETAILS));
+            auto thunder_client = std::make_shared<WPEFramework::JSONRPC::LinkType<WPEFramework::Core::JSON::IElement> >(callsign.c_str(), "");
```

Comment:
Pls create local pointer or object so that the connection is closed and recreated for every retry. Doesn't need shared pointer.